src/systems/pose_publisher/PosePublisher.cc

```diff
@@ -155,16 +155,6 @@ class gz::sim::systems::PosePublisherPrivate
   /// improves performance by avoiding memory allocation
   public: std::vector<std::pair<Entity, math::Pose3d>> staticPoses;
 
-  /// \brief A variable that gets populated with poses. This also here as a
```

Comment:
What is the reason for turning these member variables into local variables in the PosePublisherPrivate::PublishPoses() method?

Reply:
Spooky action at a distance. Described in commit message for: https://github.com/gazebosim/gz-sim/pull/2232/commits/a4c89e3ce4b6cf230ed6dd84649e0192e5d93e33

> This commit removes an "optimization" in the PosePublisher that was
causing the system to segfault when an entity was removed. In reality
this commit is unrelated to the previous few commits. However, without this change
the previous change would cause removal of an entity to crash demos that
used pose publishers. I think the crux of the issue is that protobufs takeover
memory management of objects leading to a scenario where if a reference
was freed elsewhere, the protobuf would continue to hold said reference.
When we try to do operations on the protobuf we end up double-freeing
the memory and hence there is a 💥.

Reply:
Ok, my concern is that this change didn't actually fix the problem, but just made it more unlikely to happen.

Reply:
I came to this conclusion by using the debugger. I'm pretty sure it 100% solves the problem of the crash within this plugin.